### PR TITLE
[DataGrid] Fix1295 backspace

### LINKED ISSUE #1295: [DataGrid] Loose focus when opening a textbox portal inside a cell
When we display a modal from a button included in a datagrid, we can not keep the focus inside on a TextField, it remains on the datagrid behind.
We can also delete data in row of the datagrid

## Current Behavior 😯

When you try to write in a TextField in a modal display from datagrid, you lose focus directly and you can't continue, unless you spam. If you try to delete nothing happens, or rather something happens on the datagrid behind.

## Steps to Reproduce 🕹
bug_material_ui

Bug on the modal in the datagrid, no out with the button (same component)
Steps:

1. Install 4.0.0-alpha.23
2. try Modal in datagrid

## Your Environment 🌎

  System:
    OS: Windows 10 10.0.19041

  Binaries:
    Node: 12.18.0 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 7.6.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 89.0.4389.90
    Edge: Spartan (44.19041.423.0), Chromium (89.0.774.57)   
 npmPackages:
    @material-ui/core: ^4.11.3 => 4.11.3
    @material-ui/data-grid: ^4.0.0-alpha.23 => 4.0.0-alpha.23
    @material-ui/icons: ^4.11.2 => 4.11.2
    @material-ui/lab: ^4.0.0-alpha.57 => 4.0.0-alpha.57
    @material-ui/pickers: ^3.3.10 => 3.3.10
    @material-ui/styles:  4.11.3
    @material-ui/system:  4.11.3
    @material-ui/types:  5.1.0
    @material-ui/utils:  4.11.2
    @types/react:  17.0.3
    react: ^17.0.1 => 17.0.1
    react-dom: ^17.0.1 => 17.0.1

```
  Same on Brave, Firefox
```

### PR DESCRIPTION
**Pull request by dtassone**
fix #1295, backspace issue only, lead to more issue with portals

This PR is based on #1319 as they are all related...

### CODE REVIEW

File: packages/grid/_modules_/grid/components/cell/GridCell.tsx [hunk 1]
```diff
@@ -128,10 +128,14 @@ export const GridCell: React.FC<GridCellProps> = React.memo((props) => {
...
-    if (hasFocus && cellMode === 'view' && cellRef.current) {
+    if (
+      hasFocus &&
+      cellRef.current &&
+      (!document.activeElement || !cellRef.current!.contains(document.activeElement))
+    ) {
       cellRef.current!.focus();
     }
-  });
+  }, [hasFocus]);
```
Comment (dtassone) [inline packages/grid/_modules_/grid/components/cell/GridCell.tsx:138]: fix #1295, basically we trigger focus only when hasFocus changed

**Comment by oliviertassinari** [comment]
> This PR is based on #1319 as they are all related...

They shouldn't be, no need

File: packages/grid/_modules_/grid/components/cell/GridCell.tsx (hunk 1 above)
Comment (oliviertassinari) [reply packages/grid/_modules_/grid/components/cell/GridCell.tsx:138]: This is not what I understand of the problem, this seems to be a symptom, not the root cause. From what I understand we should filter all the events that are coming from a portal in 

https://github.com/mui-org/material-ui-x/blob/728bad3ef1f481a84a042d9bb44da970b56b5c87/packages/grid/_modules_/grid/components/GridRow.tsx#L35

```diff
diff --git a/packages/grid/_modules_/grid/components/GridRow.tsx b/packages/grid/_modules_/grid/components/GridRow.tsx
index 27135564..f1a3f0be 100644
--- a/packages/grid/_modules_/grid/components/GridRow.tsx
+++ b/packages/grid/_modules_/grid/components/GridRow.tsx
@@ -27,8 +27,14 @@ export const GridRow: React.FC<RowProps> = ({ selected, id, className, rowIndex,
   const rowHeight = useGridSelector(apiRef, gridDensityRowHeightSelector);

   const publish = React.useCallback(
-    (eventName: string) => (event: React.MouseEvent) =>
+    (eventName: string) => (event: React.MouseEvent) => {
+      // Ignore portal
+      if (!event.currentTarget.contains(event.target)) {
+        return;
+      }
+
       apiRef!.current.publishEvent(eventName, apiRef?.current.getRowParams(id), event),
+    },
     [apiRef, id],
   );
```

what do you think?

Comment (oliviertassinari) [reply packages/grid/_modules_/grid/components/cell/GridCell.tsx:138]: Why would a redraw impact the focus? Maybe you are referring to what happens when we toggle between the read and edit mode, in which case, maybe we can only trigger a focus when the edit/read mode change?

**Pull request closed without merging**